libs/vec/native/src/vec/c/amd64/vec.c

```diff
@@ -0,0 +1,105 @@
+
+
+#include <emmintrin.h>
+#include <immintrin.h>
+#include <vec.h>
+
+#ifndef STRIDE_BYTES_LEN
+#define STRIDE_BYTES_LEN 32
+#endif
+
+// Horizontally add 8 singed 32-bit integers
+static inline int hsum_i32_8(const __m256i a) {
+    const __m128i sum128 = _mm_add_epi32(_mm256_castsi256_si128(a), _mm256_extractf128_si256(a, 1));
+    const __m128i hi64 = _mm_unpackhi_epi64(sum128, sum128);
+    const __m128i sum64 = _mm_add_epi32(hi64, sum128);
+    const __m128i hi32  = _mm_shuffle_epi32(sum64, _MM_SHUFFLE(2, 3, 0, 1));
+    return _mm_cvtsi128_si32(_mm_add_epi32(sum64, hi32));
+}
+
+EXPORT int stride() {
+    return STRIDE_BYTES_LEN;
+}
+
+EXPORT int dot8s(const void* a, const void* b, int dims) {
+
+	// Init accumulator(s) with 0
+	__m256i acc1 = _mm256_setzero_si256();
+
+	for(int i = 0; i < dims; i += STRIDE_BYTES_LEN) {
+
+		// Load 32 packed 8-bit integers
+		__m256i va = _mm256_loadu_si256(a + i);
+		__m256i vb = _mm256_loadu_si256(b + i);
+
+		// Multiply groups of 4 adjacent pairs of 8-bit integers in va, vb
+		// -> 4 intermediate 16-bit results.
+		// Then accumulate (+=) and store /add
+		#if __AVXVNNIINT8__
```

Comment:
Sorry for the formatting, my C editor likes tabs :/
I'll fix that and cleanup once we have decided with variant to keep (I'm leaning towards the AVX2-only, ditch the ifdefs -- but I'm unsure if we should keep a stride of 32 or double that to 64. Wdyt?)